src/winbind.c

```diff
@@ -206,15 +206,76 @@ uint32_t winbind_cli_auth(char *user, char *domain,
     return ret;
 }
 
+/*  Worst-case SID in textual representation;
+    see https://stackoverflow.com/questions/1140528/
+        what-is-the-maximum-length-of-a-sid-in-sddl-format */
+static const size_t SID_MAX_STRING_LENGTH = 184 + 1;
```

Comment:
As for using WBC_SID_STRING_BUFLEN - yes, I missed it in header-file, good point - will fix my code.

As for 190 instead of 184 - I still think that actual max SID length is 184 for ALL-DECIMAL content.
Our formulae (max_sub_authority_count * (max_str_len(sub_authority_number)+1) + max_strlen('S-1-<6-byte-authority>') + 1 /* EOL */) is 15*11 + 18 + 1 = 184 because authority is only 6 bytes (14 characters).
For some reason winbind code (see wbcSidToStringBuf) uses hexadecimal format for auth when id_auth >= UINT32_MAX, BUT it is also only up to 14 characters (2 for '0x' + 12 for 6 bytes in hex).

Bottom line - NO real reason why to use 190, wbcSidToStringBuf can't produce longer SID than calculated by us. Buffer overrun is not possible. Probably they counted authority to be 8-bytes value in some old code (counted in '25' for their formulae), but now this is not valid anymore.

Reply:
yeah I didn't think thre was an actual chance of buffer overrun 'now', but the fact they different pointed at the fact things may get out of sync in future.
Problem solved anyway, thanks